qt/applications/workbench/workbench/projectrecovery/recoverygui/projectrecoverywidgetview.py

```diff
@@ -31,15 +31,19 @@ def __init__(self, presenter, parent=None):
         self.ui.progressBar.setMinimum(0)
         self._add_data_to_table()
 
+        # keep copy of reference to the code editor
+        self.editor = None
+
     def reject(self):
         self.presenter.start_mantid_normally()
 
     def set_progress_bar_maximum(self, new_value):
         self.ui.progressBar.setMaximum(new_value)
 
     def connect_progress_bar(self):
-        self.presenter.project_recovery.loader.multi_file_interpreter.current_editor().sig_progress.connect(
-            self.update_progress_bar)
+        self.editor = self.presenter.project_recovery.loader.multi_file_interpreter.current_editor()
+        self.editor.sig_progress.connect(self.update_progress_bar)
```

Comment:
Can we do this in a single call? Something like

```python
self.editor.connect_editor_to_sig_process(self.update_progress_bar)
```

where we pass the callable in and it connects it for us?

Reply:
Done :+1: